### DISCUSSION TITLE
Multiple mime type request bodies inaccessible from UI

### DISCUSSION THREAD

**Discussion by asuthy** [scalar/scalar#1117] [Q&A] [answered]
OpenAPI YAML defining a simple post endpoint that supports both JSON and XML request bodies

```
openapi: 3.0.3
info:
  title: API
  version: 1.0.0
servers:
  - url: http://localhost:55321

paths:
  /record:

    post:
      summary: Create a new record
      requestBody:
        description: JSON
        content:
          application/json:
            schema:
              type: object
              properties:
                id:
                  type: integer
                  format: int64
                  example: 1
                title:
                  type: string
                  example: The record title
          application/xml:
            schema:
              type: object
              properties:
                id:
                  type: integer
                  format: int64
                  example: 1
                title:
                  type: string
                  example: The record title
              xml:
                name: record
        required: true
      responses:
        '201':
          description: Record Created
          content:
            application/json:
              schema:
                type: object
                properties:
                  id:
                    type: integer
                    example: 1
                  title:
                    type: string
                    example: The record title
                        
      security:
        - bearerAuth: []

components:

  securitySchemes:
    bearerAuth:
      type: http
      scheme: bearer
      bearerFormat: JWT

```

Rendered in Swagger UI, this provides the option to change the request body mime type, see the example data in the relevant type, and test calling the endpoint with the selected request body type ...

![SwaggerEditor](https://github.com/scalar/scalar/assets/9988838/7fd8cbde-8d00-483d-a90f-1891e7b468bf)

![SwaggerEditor](https://github.com/scalar/scalar/assets/9988838/ceae3a9c-1eea-4ccf-a003-7614208b65bd)

![SwaggerEditor](https://github.com/scalar/scalar/assets/9988838/5af7e7b0-e7e6-4510-a1bd-e778ec5ff7d3)

![SwaggerEditor](https://github.com/scalar/scalar/assets/9988838/f519d9d4-ef41-44fb-a284-2e141527b6b2)

Rendered in the API Reference, a dropdown is provided to select the request body type, but this seems to have no effect.

![Swagger_Editor_—_Preview__Edit_and_Generate_Swagger_Documentation](https://github.com/scalar/scalar/assets/9988838/328284d8-830f-4086-9e1d-6c96d425adaa)

![Swagger_Editor_—_Preview__Edit_and_Generate_Swagger_Documentation](https://github.com/scalar/scalar/assets/9988838/96084139-54dc-46f7-87ba-3a0ad2578b38)

If the JSON request body is excluded from the definition, then the XML request body is actually picked up and can be tested.

![Swagger_Editor_—_Preview__Edit_and_Generate_Swagger_Documentation](https://github.com/scalar/scalar/assets/9988838/d61934a5-7b5d-468d-9b23-f300339e012d)

Maybe I'm doing something wrong?

Digging around the code led me here, which looked to be preferring a JSON example if one was defined, and only falling back to different request body types if there was no JSON example? https://github.com/scalar/scalar/blob/main/packages/api-reference/src/helpers/getRequestBodyFromOperation.ts

**Comment by hanspagel** [comment, marked as answer]
No, you’re not doing it wrong, we need to fix that! I’ve created an issue: #1118

**Reply by asuthy** [reply]
Thanks @hanspagel
